props/saber_fett263_buttons.h

```diff
@@ -293,248 +660,3355 @@ class SaberFett263Buttons : public virtual PropBase {
+            style_revert_ = style_num_;               
+            PlayMenuSound("mselect.wav");
+            break;
+  #else
+            menu_type_ = MENU_BLADE_STYLE;
+            blade_num_ = 0;
+            PlayMenuSound("mblade.wav");
+            break;
+  #endif
+          case 2:
+            if (style_parser.MaxUsedArgument(current_preset_.GetStyle(1)) == 0) {
+              menu_type_ = MENU_COLOR;
+              PlayMenuSound("mselect.wav");
+              color_revert_ = SaberBase::GetCurrentVariation();
+              ToggleColorChangeMode();
+              break;
+            } else {
+  #if NUM_BLADES == 1
+            blade_num_ = 1;
+            menu_type_ = MENU_COLOR_SUB;
+            menu_sub_pos_ = 0;
+            PlayMenuSound("mcolorsb.wav");
+            effect_num_ = 0;
+            break;
+  #else
+            menu_type_ = MENU_BLADE_COLOR;
+            blade_num_ = 0;
+            PlayMenuSound("mblade.wav");
+            break;                
+  #endif
+            }
+            break; 
+          case 3:
+            menu_type_ = MENU_FONT;
+            font_num_ = 0;                                               
+            num_fonts_ = RunCommandAndGetSingleLine("list_fonts", nullptr, 0, 0, 0);
+            if (!restore_point.get()) restore_point = std::move(current_preset_.font);
+            PlayMenuSound("mselect.wav");
+            break;
+          case 4:
+            num_tracks_ = RunCommandAndGetSingleLine("list_current_tracks", nullptr, 0, 0, 0);
+            StartOrStopTrack();
+            menu_type_ = MENU_TRACK;
+            track_num_ = 0;
+            if (!restore_point.get()) restore_point = std::move(current_preset_.track);
+            PlayMenuSound("mselect.wav");
+            break;
+          case 5:
+            menu_type_ = MENU_SETTING_SUB;
+            PlayMenuSound("moption.wav");
+            menu_sub_pos_ = 0;
+            break;
+          case 6:
+            menu_type_ = MENU_COPY;
+            PlayMenuSound("mconfirm.wav");
+            break;
+          case 7:
+            menu_type_ = MENU_DELETE;
+            PlayMenuSound("mconfirm.wav");
+            break;
+          case 8:
+            PlayMenuSound("mselect.wav");
+            MenuExit();
+            break;
+        }
+        break;
+  #if NUM_BLADES > 1          
+      case MENU_BLADE_STYLE:
+        if (blade_num_ == 0) {
+          PlayMenuSound("mblade.wav");
+          break;
+        }
+        show_preview_.Stop(blade_preview_);
+        blade_preview_ = 0;
+        menu_type_ = MENU_STYLE;
+        num_presets_ = current_config->num_presets;
+        tmp = current_preset_.GetStyle(blade_num_);
+        style_num_ = FirstWord(tmp, "builtin") ? atoi(SkipWord(tmp)) : 0; 
+        style_revert_ = style_num_;               
+        MenuSelect();
+        break;
+      case MENU_BLADE_COLOR:
+  #if NUM_BLADES > 2
+        menu_type_ = MENU_COLOR_SUB;
```

Comment:
Separate out the parts that are the same outside of the #if/#else/#endif.
The menu_type_ is the same, the break at the end is the same, etc.

Reply:
Updated